utils/src/tap_error_log.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+pub trait LogError
+where
+    Self: Sized,
+{
+    fn log_err(self) -> Self;
```

Comment:
I'm not sure how useful it is to log errors without any context. I think that usually it would be hard to figure out what's wrong only by the "X error occurred" message.

Reply:
You have to think about this in a hierarchical form. When someone submits a block, everything will return an error, so you can see where the chain of errors started. With our design of errors with thiserror, we should have all the information we need, assuming we use `log_err()` over every returned error.